File: DSA_CSHARP/Matrices/ToeplitzMatrix.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Reflection.Metadata;
using System.Text;
using System.Threading.Tasks;
using System.Xml.Linq;

namespace DSA_CSHARP.Matrices;
internal class ToeplitzMatrix
{
    //A Toeplitz matrix(also called a diagonal-constant matrix) is a matrix in which each descending diagonal from left to right is constant.
    //This means that all elements along a diagonal have the same value.

    static void ReadMatrix(int[,] matrix, int n)
    {
        Console.WriteLine("Enter elements for the Toeplitz Matrix:");

        // First row
        for (int j = 0; j < n; j++)
        {
            Console.Write($"Element [0][{j}]: ");
            matrix[0, j] = int.Parse(Console.ReadLine());
        }

        // First column (excluding [0][0])
        for (int i = 1; i < n; i++)
        {
            Console.Write($"Element [{i}][0]: ");
            matrix[i, 0] = int.Parse(Console.ReadLine());
        }

        // Fill remaining values using Toeplitz property
        for (int i = 1; i < n; i++)
        {
            for (int j = 1; j < n; j++)
            {
                matrix[i, j] = matrix[i - 1, j - 1];
            }
        }
    }

    static void PrintMatrix(int[,] matrix, int n)
    {
        Console.WriteLine("\nToeplitz Matrix Output:");
        for (int i = 0; i < n; i++)
        {
            for (int j = 0; j < n; j++)
            {
                Console.Write(matrix[i, j] + " ");
            }
            Console.WriteLine();
        }
    }

    public string Main()
    {
        Console.Write("Enter the size of the square matrix (n x n): ");
        int n = int.Parse(Console.ReadLine());

        int[,] matrix = new int[n, n];
        ReadMatrix(matrix, n);
        PrintMatrix(matrix, n);

        return string.Empty;
    }


}

File: DSA_PYTHON/Matrices/ToeplitzMatrix.py

class ToeplitzMatrix:
    # A Toeplitz matrix(also called a diagonal-constant matrix) is a matrix in which each descending diagonal from left to right is constant.
    # This means that all elements along a diagonal have the same value.

    def read_matrix(self, n):
        matrix = [[0 for _ in range(n)] for _ in range(n)]
        print("Enter elements for Toeplitz Matrix:")

        # First row
        for j in range(n):
            matrix[0][j] = int(input(f"Element [0][{j}]: "))

        # First column (excluding [0][0])
        for i in range(1, n):
            matrix[i][0] = int(input(f"Element [{i}][0]: "))

        # Fill rest using diagonal rule
        for i in range(1, n):
            for j in range(1, n):
                matrix[i][j] = matrix[i - 1][j - 1]

        return matrix

    def print_matrix(self, matrix):
        print("\nToeplitz Matrix Output:")
        for row in matrix:
            for val in row:
                print(val, end=" ")
            print()

    def main(self):
        n = int(input("Enter the size of the square matrix (n x n): "))
        matrix = self.read_matrix(n)
        self.print_matrix(matrix)
